src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigController.java

```diff
@@ -22,316 +22,83 @@
-        } catch (AlertException e) {
-            logger.error(e.getMessage(), e);
-            return responseFactory.createMessageResponse(HttpStatus.INTERNAL_SERVER_ERROR, "There was an issue retrieving data from the database for ID: " + id);
-        }
-
-        if (optionalModel.isPresent()) {
-            JobFieldModel fieldModel = optionalModel.get();
-            boolean hasPermissions = hasRequiredPermissions(fieldModel.getFieldModels(), authorizationManager::hasReadPermission);
-            if (!hasPermissions) {
-                return responseFactory.createForbiddenResponse();
-            }
-            return responseFactory.createOkContentResponse(contentConverter.getJsonString(fieldModel));
-        }
-
-        return responseFactory.createNotFoundResponse("Configuration not found for the specified id");
-    }
-
-    @PostMapping
-    public ResponseEntity<String> postConfig(@RequestBody(required = true) JobFieldModel restModel) {
-        boolean hasPermissions = hasRequiredPermissions(restModel.getFieldModels(), authorizationManager::hasCreatePermission);
-        if (!hasPermissions) {
-            return responseFactory.createForbiddenResponse();
-        }
-
-        try {
-            return runPostConfig(restModel);
-        } catch (AlertException e) {
-            logger.error(e.getMessage(), e);
-            return responseFactory.createInternalServerErrorResponse(restModel.getJobId(), e.getMessage());
-        }
-    }
-
-    private ResponseEntity<String> runPostConfig(JobFieldModel restModel) throws AlertException {
-        String id = restModel.getJobId();
-        if (StringUtils.isNotBlank(id) && jobConfigActions.doesJobExist(id)) {
-            return responseFactory.createConflictResponse(id, "Provided id must not be in use. To update an existing configuration, use PUT.");
-        }
-
-        try {
-            JobFieldModel updatedEntity = jobConfigActions.saveJob(restModel);
-            return responseFactory.createCreatedResponse(updatedEntity.getJobId(), "Created");
-        } catch (AlertFieldException e) {
-            return responseFactory.createFieldErrorResponse(id, "There were errors with the configuration.", e.getFieldErrors());
-        }
-    }
-
-    @PutMapping("/{id}")
-    public ResponseEntity<String> putConfig(@PathVariable UUID id, @RequestBody(required = true) JobFieldModel restModel) {
-        boolean hasPermissions = hasRequiredPermissions(restModel.getFieldModels(), authorizationManager::hasWritePermission);
-        if (!hasPermissions) {
-            return responseFactory.createForbiddenResponse();
-        }
-
-        try {
-            return runPutConfig(id, restModel);
-        } catch (AlertException e) {
-            logger.error(e.getMessage(), e);
-            return responseFactory.createInternalServerErrorResponse(restModel.getJobId(), e.getMessage());
-        }
+    @GetMapping("/validate")
+    public List<JobFieldStatuses> validateJobs() {
```

Comment:
Will this clash with the overridden validate method?

Reply:
no because the overridden validate method is a POST mapping and takes a parameter.  The method signature is different and has a different request mapping associated with it.  They do not conflict.